File: ant-sdk/antd-java/src/main/java/com/autonomi/antd/models/DataPutResult.java
package com.autonomi.antd.models;

/**
 * Result of a private data put. The DataMap is returned to the caller; it is
 * NOT stored on-network. Both the REST and gRPC transports populate
 * {@code chunksStored} and {@code paymentModeUsed}.
 *
 * @param dataMap         hex-encoded caller-held DataMap
 * @param chunksStored    number of chunks stored on the network
 * @param paymentModeUsed which payment mode was actually used: "auto", "merkle", or "single"
 */
public record DataPutResult(
        String dataMap,
        long chunksStored,
        String paymentModeUsed) {
    public DataPutResult(String dataMap) {
        this(dataMap, 0L, "");
    }
}

File: ant-sdk/antd-php/src/Models/DataPutResult.php
<?php

declare(strict_types=1);

namespace Autonomi\Antd\Models;

/**
 * Result of a private data put. The DataMap is returned to the caller; it
 * is NOT stored on-network.
 */
readonly class DataPutResult
{
    public function __construct(
        /** Hex-encoded caller-held DataMap. */
        public string $dataMap,
        /** Number of chunks stored on the network. */
        public int $chunksStored,
        /** Which payment mode was actually used: "auto", "merkle", or "single". */
        public string $paymentModeUsed,
    ) {}
}
